### PR TITLE
add support for selecting text

### PR DESCRIPTION
**Pull request by karengu**
language-selector

### CODE REVIEW

File: tinymce-language-selector/plugin.js [hunk 1]
```diff
@@ -6,17 +6,20 @@ tinymce.PluginManager.add('language', function(editor) {
...
     const selectedNode = editor.selection.getNode();
     const newText = editor.selection.getContent({ format: 'text' });
-    const contents = newText || '&#65279';
+    const contents = newText || '&#65279'; // use zero-width character as placeholder if no existing text
     const newSpanText = lang === BROWSER_DEFAULT ? `<span id="new_span">${contents}</span>` : `<span lang="${lang}" id="new_span">${contents}</span>`;
     // may be in the middle of the span, so split it into two
     if (selectedNode.nodeName === 'SPAN') {
       if (selectedNode.lang) {
-        editor.execCommand('mceReplaceContent', false, `</span>${newSpanText}<span lang="${selectedNode.lang}">`);
+        editor.selection.setContent(''); // delete any existing text
```
Comment (Dillon-Dumesnil) [inline tinymce-language-selector/plugin.js]: This should just be moved outside of the if conditions if we are going to call it in every case.

File: tinymce-language-selector/plugin.js [hunk 2]
```diff
@@ -6,17 +6,20 @@ tinymce.PluginManager.add('language', function(editor) {
...
+        editor.execCommand('mceInsertRawHTML', false, `</span>${newSpanText}<span lang="${selectedNode.lang}">`);
```
Comment (Dillon-Dumesnil) [inline tinymce-language-selector/plugin.js:15]: So what's the difference between the mceReplaceContent and mceInsertRawHTML?

**Review by Dillon-Dumesnil — COMMENTED** [review]
What is the docs/language-selector.gif?

I think it can be a bit more DRY, but looks good overall. Just want a little more context on it.

**Comment by karengu** [comment]
> What is the docs/language-selector.gif?

It's just for the README.

File: tinymce-language-selector/plugin.js (hunk 2 above)
Comment (karengu) [reply tinymce-language-selector/plugin.js:15]: `mceReplaceContent` replaces what is already selected, whereas `mceInsertRawHTML` inserts the HTML at the cursor point, which is what we want since we need to close any existing spans.

File: tinymce-language-selector/plugin.js [hunk 3]
```diff
@@ -131,6 +129,17 @@ tinymce.PluginManager.add('language', function(editor) {
...
         updateButtonText(selectedLang);
       };
+      editor.addShortcut('Meta+L', 'Switch to default language', () => {
+        const selectedNode = editor.selection.getNode();
+        const currentLang = selectedNode.lang ? selectedNode.lang : BROWSER_DEFAULT;
+        if (currentLang !== BROWSER_DEFAULT) {
+          editor.undoManager.transact(() => {
+            replaceText(BROWSER_DEFAULT);
+            buttonApi.setActive(false);
+            updateCurrentLanguage();
```
Comment (Dillon-Dumesnil) [inline tinymce-language-selector/plugin.js:161]: Above you didn't call this and now you are. Why the change?

File: tinymce-language-selector/plugin.js [hunk 4]
```diff
@@ -35,6 +36,13 @@ tinymce.PluginManager.add('language', function(editor) {
 
   const openDialog = (buttonApi) => {
     const selectedNode = editor.selection.getNode();
+    if (['OL', 'UL'].includes(selectedNode.nodeName)) {
```
Comment (Dillon-Dumesnil) [inline tinymce-language-selector/plugin.js:81]: What happened here? Wasn't this one of the features you showed off in the gif?

**Review by Dillon-Dumesnil — APPROVED** [review]
Couple clarifying questions, but otherwise, looks good.

File: tinymce-language-selector/plugin.js (hunk 3 above)
Comment (karengu) [reply tinymce-language-selector/plugin.js:161]: I thought that the `keyup` hook would catch this case, but if you Cmd-L out of a language, it doesn't update the button text until you keep typing. (This should've been in the previous PR but I missed it.)

File: tinymce-language-selector/plugin.js (hunk 4 above)
Comment (karengu) [reply tinymce-language-selector/plugin.js:81]: It works if you don't have anything highlighted, but if you do have things highlighted, it gets really hard to properly assign the spans, since you can have only a few items of the list highlighted, the list items can have their own formatting, etc.
